src/main/java/org/apache/xml/security/encryption/XMLCipherInput.java

```diff
@@ -115,7 +115,14 @@ public void setSecureValidation(boolean secureValidation) {
             try {
                 ResourceResolverContext resolverContext =
                     new ResourceResolverContext(uriAttr, null, secureValidation);
-                input = ResourceResolver.resolve(resolverContext);
+                if (resolverContext.isURISafeToResolve()) {
+                    input = ResourceResolver.resolve(resolverContext);
+                } else {
+                    String uriToResolve = uriAttr != null ? uriAttr.getValue() : null;
+                    Object[] exArgs = {uriToResolve != null ? uriToResolve : "null", null};
```

Comment:
I don't think you need to set the first argument to `"null"` if `uriToResolve` is `null`. I think the exception message should be able to handle `null` arguments, i.e. `MessageFormat.format` should print `"null"` for `null` arguments.